src/main/java/com/karankumar/bookproject/backend/entity/Publisher.java

```diff
@@ -0,0 +1,47 @@
+package com.karankumar.bookproject.backend.entity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Data
+@JsonIgnoreProperties(value = {"id", "books"})
+@EqualsAndHashCode(callSuper = true, exclude= "books")
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Publisher extends BaseEntity{
+    
+    @NotNull
+    @NotEmpty
```

Comment:
Use `@NotBlank`? Also can you please add a test for this?

Reply:
Done